content/redirect.js

```diff
@@ -2,10 +2,26 @@ const fs = require("fs");
 const path = require("path");
 
 const { resolveFundamental } = require("../libs/fundamental-redirects");
-const { CONTENT_ROOT, VALID_LOCALES } = require("./constants");
+const {
+  CONTENT_ROOT,
+  CONTENT_TRANSLATED_ROOT,
+  FORBIDDEN_URL_SYMBOLS,
+  VALID_LOCALES,
+} = require("./constants");
+
+const FORBIDDEN_URL_SYMBOLS_SET = new Set(FORBIDDEN_URL_SYMBOLS);
+
+function urlHasInvalidSymbols(url) {
+  return url.split("").some((c) => FORBIDDEN_URL_SYMBOLS_SET.has(c));
+}
```

Comment:
Nit: Instead of a function here and then throwing elsewhere, unable to say what's wrong, how about something like:

```javascript
function checkURLInvalidSymbols(url) {
  for (const character of FORBIDDEN_URL_SYMBOLS) {
    if (url.includes(character)) {
       throw new Error(`URL contains invalid character '${character}'`);
    }
  }
}
```
that way the error will be a lot more useful when thrown. And the checking-code will be closer to the error handling.